test/integration/build_admission_test.go

```diff
@@ -23,7 +35,7 @@ func TestPolicyBasedRestrictionOfBuildCreateAndCloneByStrategy(t *testing.T) {
 	builds := map[string]*buildapi.Build{}
 
 	// Create builds to setup test
-	for _, strategy := range buildStrategyTypes() {
+	for _, strategy := range buildStrategyTypesDefault() {
```

Comment:
I think you could just add a couple things to the beginning of the original test and keep the remainder of the file unmodified:

New: ensure restricted types fail, unrestricted types succeed
New: Grant restricted
… continue with original test case:
Ensure all types succeed
Remove permissions
Ensure all types fail